postfix/init.sls

```diff
@@ -17,33 +17,21 @@ postfix:
 
 # manage /etc/aliases if data found in pillar
 {% if 'aliases' in pillar.get('postfix', '') %}
-  {%- set need_newaliases = False %}
-  {%- set file_path = postfix.aliases_file %}
-  {%- if ':' in file_path %}
-    {%- set file_type, file_path = postfix.aliases_file.split(':') %}
-  {%- else %}
-    {%- set file_type = default_database_type %}
-  {%- endif %}
-  {%- if file_type in ("btree", "cdb", "dbm", "hash", "sdbm") %}
-    {%- set need_newaliases = True %}
-  {%- endif %}
-postfix_alias_database:
-  file.managed:
-    - name: {{ file_path }}
-    - source: salt://postfix/aliases
-    - user: root
-    - group: root
-    - mode: 644
-    - template: jinja
-    - require:
-      - pkg: postfix
-  {%- if need_newaliases %}
-  cmd.wait:
-    - name: newaliases
-    - cwd: /
-    - watch:
-      - file: {{ file_path }}
-  {%- endif %}
+{%- if 'present' in pillar.get('postfix', '')['aliases'] %}
+{%- for user, target in pillar.get('postfix', '')['aliases']['present'].items() %}
+{{ 'postfix_alias_' ~ user ~ '_' ~ loop.index0 }}:
```

Comment:
`dict` cannot have multiple keys with the same name so you might want to use the user part as it would allow someone to put a require on it rather than try to depend on a loop index that might change by adding or removing entries unrelated to the actual user.